projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -214,6 +213,16 @@ def ingest_payload(
         """
         pass
 
+    def process_payload(self, payload: List[dict]) -> List[dict]:
```

Comment:
My understanding is, hook implementations chaining is already facilitated by pluggy. Does pluggy support passing the parameter from a returned value from a previous execution? Otherwise, I would suggest a param referencing a mutable object, encapsulated within the the scope of the chaining method.

Reply:
Note that we are not using pluggy here. But our own (called router) extensibility. 

This is because pluggy does not support conditional hooks (hooks activated only if certain configuration is set for example HTTP ingest hook used only if ingest_method is HTTP for example)

Reply:
As @tozka has mentioned above, we do not rely on pluggy here. With the current plugin configuration, we can set a plugin to be placed at the beginning of the execution queue, the end, or somewhere in between, but we cannot set it to be executed at a specific point before or after another plugin.

As for your other point, could you elaborate what you mean by param referencing?

Reply:
> Otherwise, I would suggest a param referencing a mutable object, encapsulated within the the scope of the chaining method.

@ivakoleva could you clarify what you mean here?

Reply:
Discussed offline. Thanks